packages/stencil-library/src/components/icon/icon.e2e.ts

```diff
@@ -0,0 +1,150 @@
+        expect(functionalUseElement).toEqualAttribute("xlink:href", functionalIconSrc)
+
+        const decorativeSVG = await renderAndGetSVGElement(`
+            <njwds-icon decorative icon="mail"></njwds-icon>
+        `)
+        const decorativeUseElement = await decorativeSVG.find("use")
+        const decorativeIconSrc = getE2EPageAssetPath("img/sprite.svg#mail")
+        expect(decorativeUseElement).toEqualAttribute("xlink:href", decorativeIconSrc)
+    })
+
+
+    it("has an SVG with the 'img' role whether or not the icon is decorative", async () => {
+        const functionalSVG = await renderAndGetSVGElement(`
+            <njwds-icon icon="mail" icon-title="mail"></njwds-icon>
+        `)
+        const decorativeSVG = await renderAndGetSVGElement(`
+            <njwds-icon icon="mail" decorative></njwds-icon>
+        `)
+        expect(functionalSVG).toEqualAttribute("role", "img")
+        expect(decorativeSVG).toEqualAttribute("role", "img")
+    })
+
+    it("has an non-focusable SVG, whether or not the icon is decorative", async () => {
+        const functionalSVG = await renderAndGetSVGElement(`
+            <njwds-icon icon="mail" icon-title="mail"></njwds-icon>
+        `)
+        expect(functionalSVG).toEqualAttribute("focusable", "false")
+
+        const decorativeSVG = await renderAndGetSVGElement(`
+            <njwds-icon icon="mail" decorative></njwds-icon>
+        `)
+        expect(decorativeSVG).toEqualAttribute("focusable", "false")
+    })
+
+    describe("functional icons (decorative='false')", () => {
+        it("does not have the aria-hidden attribute", async () => {
+            const svgElement = await renderAndGetSVGElement(`
+                <njwds-icon icon="mail" icon-title="Get updates"></njwds-icon>
+            `)
+
+            expect(svgElement).not.toHaveAttribute("aria-hidden")
+        })
+
+        it("contains a title element which labels the SVG", async () => {
+            const svgElement = await renderAndGetSVGElement(`
+                <njwds-icon icon="mail" icon-title="Get updates"></njwds-icon>
+            `)
+
+            const titleElement = await svgElement.find("title")
+            expect(titleElement).toEqualText("Get updates")
+
+            const titleElementId = titleElement.getAttribute("id")
+            const svgLabelledBy = svgElement.getAttribute("aria-labelledby")
+
+            expect(svgLabelledBy).toBe(titleElementId)
+        })
+
+        it("if no icon title is provided, set it to the icon name", async () => {
+            /* We throw an error when no icon title is provided on a functional icon (see spec tests), but if the user still doesn't provide a title, we provide a reasonable placeholder. */
+            suppressConsoleErrorFromE2EPage()
+
+            const svgElement = await renderAndGetSVGElement(`
+                <njwds-icon icon="mail"></njwds-icon>
+            `)
+            const titleElement = await svgElement.find("title")
+            try {
```

Comment:
[dust] What's the the purpose of this try/catch block?

Reply:
Ah good call. I kept the tests you wrote before, but should have checked this a bit closer. TY